internal/common/rpc/error_wrapper.go

```diff
@@ -40,7 +40,7 @@ var (
 )
 
 // NewWorkflowServiceErrorWrapper creates a new wrapper to WorkflowService that will convert gRPC status errors to service errors.
-func NewWorkflowServiceErrorWrapper(service workflowservice.WorkflowServiceClient) *workflowServiceErrorWrapper {
+func NewWorkflowServiceErrorWrapper(service workflowservice.WorkflowServiceClient) workflowservice.WorkflowServiceClient {
```

Comment:
[This](https://github.com/golang/go/wiki/CodeReviewComments) says:

> The implementing package should return concrete (usually pointer or struct) types: that way, new methods can be added to implementations without requiring extensive refactoring.

I found this rule useful and try to follow it whenever it is possible.